cypress/e2e/agreements/agreement-line-create-multiple.cy.js

```diff
@@ -0,0 +1,234 @@
+    console.log('createUser');
+    cy.createUserWithPwAndPerms(editUser, editPermissions);
+    cy.createUserWithPwAndPerms(viewUser, viewPermissions);
+
+    console.log('do not hide internal agreements knowledgebase');
+    cy.getAdminToken();
+    cy.getAgreementsGeneralSettings().then((settings) => {
+      if (settings?.hideEResourcesFunctionality === true) {
+        isChecked = false;
+        cy.setAgreementsGeneralSettings({ hideEResourcesFunctionality: false });
+      }
+    });
+    cy.login(Cypress.env('login_username'), Cypress.env('login_password'));
+    cy.getAdminToken();
+    LocalKBAdminAppInteractor.openLocalKbAdminApp();
+    LocalKBAdminAppInteractor.uploadJsonFileAndAwaitCompletion(fileName);
+    AgreementAppInteractor.openAgreementsApp();
+    AgreementAppInteractor.createAgreement(agreement);
+    AgreementAppInteractor.openLocalKB();
+    AgreementAppInteractor.searchForPackage(packageName);
+    PackageViewInteractor.addToBasket();
+    cy.logout();
+  });
+
+  after(() => {
+    console.log('delete users');
+    cy.getAdminToken();
+    cy.deleteUserViaApi(editUser.userId);
+    cy.deleteUserViaApi(viewUser.userId);
+
+    console.log('delete agreement line and agreement');
+    cy.login(Cypress.env('login_username'), Cypress.env('login_password'));
+    cy.visit('/erm/agreements');
+    AgreementAppInteractor.searchAgreement(agreementName);
+    AgreementViewInteractor.paneExists(agreementName);
+    // Potential reformatting
+    deleteFirstAgreementLine();
+    deleteFirstAgreementLine();
+    AgreementViewInteractor.delete(agreementName);
+    cy.logout();
+
+    console.log(
+      'set hideEResourcesFunctionality checkbox back to its original state'
+    );
+    if (isChecked === true) {
+      cy.setAgreementsGeneralSettings({ hideEResourcesFunctionality: true });
+    }
+  });
+
+  describe('user actions', () => {
+    describe('editUser actions', () => {
+      before(() => {
+        cy.login(editUser.username, editUser.password);
+        cy.getToken(editUser.username, editUser.password);
+      });
+
+      after(() => {
+        cy.logout();
+      });
+
+      it('should open the agreements app', () => {
+        AgreementAppInteractor.openAgreementsApp();
+      });
+
+      it('select agreement from results list', () => {
+        AgreementAppInteractor.searchAgreement(agreementName);
+        AgreementViewInteractor.paneExists(agreementName);
+      });
+
+      it('expand agreement lines accordion in agreement view', () => {
+        AgreementViewInteractor.openAgreementLinesAccordion();
+      });
+
+      it('open agreement line accordion actions and click new agreement', () => {
+        AgreementViewInteractor.newAgreementLine();
+      });
+
+      it('select e-resource from selection', () => {
+        AgreementLineFormInteractor.selectEresource(packageName);
+      });
+
+      it('link e-resource from selection', () => {
+        AgreementLineFormInteractor.linkSelectedEresource();
```

Comment:
This test fails for me, it says:

```(uncaught exception)TypeError: this.selectList.contains is not a function
(uncaught exception)TypeError: this.selectList.contains is not a function
TypeError
The following error originated from your application code, not from Cypress.

> this.selectList.contains is not a function

When Cypress detects uncaught errors originating from your application it will automatically fail the current test.

This behavior is configurable, and you can choose to turn this off by listening to the uncaught:exception event.

https://on.cypress.io/uncaught-exception-from-application

at SingleSelect.hideOnBlur (http://localhost:3000/bundle.js:226047:30)
at SingleSelect.handleControlBlur (http://localhost:3000/bundle.js:226025:10)
at HTMLUnknownElement.callCallback (http://localhost:3000/bundle.js:515888:14)
at Object.invokeGuardedCallbackDev (http://localhost:3000/bundle.js:515937:16)
at invokeGuardedCallback (http://localhost:3000/bundle.js:515999:31)
at invokeGuardedCallbackAndCatchFirstError (http://localhost:3000/bundle.js:516013:25)
at executeDispatch (http://localhost:3000/bundle.js:520186:3)
at processDispatchQueueItemsInOrder (http://localhost:3000/bundle.js:520218:7)
at processDispatchQueue (http://localhost:3000/bundle.js:520231:5)
at dispatchEventsForPlugins (http://localhost:3000/bundle.js:520242:3)
```

All following tests may fail because of this one.
Does this test pass for you?